eppo/src/main/java/cloud/eppo/android/EppoClient.java

```diff
@@ -160,9 +161,9 @@ protected EppoValue getTypedAssignment(String flagKey, String subjectKey, Subjec
         }
 
         if (assignedValue != null && assignmentLogger != null && evaluationResult.doLog()) {
-            String experimentKey = evaluationResult.getFlagKey();
-            String variationKey = evaluationResult.getVariation().getKey();
             String allocationKey = evaluationResult.getAllocationKey();
+            String experimentKey = flagKey + '-' + allocationKey; // Our experiment key is derived by hyphenating the flag key and allocation key
```

Comment:
I decided to just inline this vs. a separate function in a separate file since it was one line and only used here.